### ISSUE TITLE
Training multiple subjects

### ISSUE THREAD

**Issue by JoePenna** [XavierXiao/Dreambooth-Stable-Diffusion#12]
I've seen this mentioned in some other threads here, but figured I'd start a new issue.

This is what my trained model looked like before:
![B63FC792-1BFF-40DB-9F8A-D82D766428EF](https://user-images.githubusercontent.com/100188076/189505949-ade1c630-e33b-4bcc-96b9-f4511d60da8c.png)

And after training someone else:
![CF3B4E97-16EA-460B-8CB5-484DCDF4C2C8](https://user-images.githubusercontent.com/100188076/189505961-3f7b8a1a-55cf-4049-a2a3-1888ba89a97e.png)

Every setting was different for the second person (class word, regularization images, etc). Yet there's certainly some sort of melding happening to the original trained token. Second token looked mostly fine (though there were some of the cfg_scale colorful artifacts).

Does anyone have any info on how to train multiple tokens at once (which seems possible, when looking deep in the code)?

Or any info on how to preserve the original trained token while adding another?

**Comment by binaryninja** [comment]
When training the second person did you change the name of the hardcoded token "sks"?

https://github.com/XavierXiao/Dreambooth-Stable-Diffusion/blob/main/ldm/data/personalized.py#L10

**Comment by ExponentialML** [comment]
> When training the second person did you change the name of the hardcoded token "sks"?
> 
> https://github.com/XavierXiao/Dreambooth-Stable-Diffusion/blob/main/ldm/data/personalized.py#L10

This happens even when changing the hardcoded token. There's a problem somewhere that's pushing all embeddings to the same space during training. I thought it was the global seed, but that isn't the case.

**Comment by JoePenna** [comment]
@binaryninja @ExponentialML absolutely, I changed every single thing I could. Including the training prompts.

**Comment by ExponentialML** [comment]
> @binaryninja @ExponentialML absolutely, I changed every single thing I could. Including the training prompts.

I haven't tried this as I'm a bit busy this week, but this could be a possible solution.

After training, prune the model checkpoints from 11GB to 2GB using this script. This can be done before or after training presumably:
https://github.com/harubaru/waifu-diffusion/blob/main/scripts/prune.py

Then, merge all of the model checkpoints as stated in this repository:
https://github.com/Jack000/glid-3-xl-stable#trainingfine-tuning

I don't know what effect it would have, but this might be a step in the right direction if there aren't any viable fixes to the problem.

**Comment by thegenerativegeneration** [comment]
I had a similar idea, but for some reason only one of the people I use images to train of ends up in the model, even though all of them are included in the dataset. Still figuring out why.

**Comment by ETA-22** [comment]
@ExponentialML How do you do multiple classes for regularizing these diverse classes as well? Like say I have "Woman" "Man" "Cat" "Dog" pools of regularization images, then have the new training images "Jane" "John" "Buddy" and "Ollie" to train into each of those classes.

How do I match them to their respective classes in the same model training session?
